### PR TITLE
Package Vendor

### PR DESCRIPTION
**Pull request by twokul**
This change is related to the [spike](https://github.com/ember-cli/ember-cli/pull/7562) and is backwards compatible. To avoid making changes to `ember-app` in one go, we can move processing/packaging logic (one tree at a time) to `DefaultPackager` while adding tests/documentation and deprecating private undocumented methods.

### CODE REVIEW

File: lib/broccoli/default-packager.js [hunk 1]
```diff
@@ -44,12 +46,55 @@ module.exports = class DefaultPackager {
   packageBower(tree, bowerDirectory) {
     if (this._cachedBower === null) {
       this._cachedBower = new Funnel(tree, {
-        srcDir: '/',
```
Comment (Turbo87) [inline lib/broccoli/default-packager.js]: seems unrelated?

File: tests/unit/broccoli/default-packager/vendor-test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  after(co.wrap(function *() {
+    yield input.dispose();
+  }));
+
+  it('caches packaged vendor tree', co.wrap(function *() {
+    let defaultPackager = new DefaultPackager();
+
+    expect(defaultPackager._cachedVendor).to.equal(null);
+
+    yield buildOutput(defaultPackager.packageVendor(input.path()));
```
Comment (Turbo87) [inline tests/unit/broccoli/default-packager/vendor-test.js]: I think we might have to `output.dispose()`, right?

File: lib/broccoli/default-packager.js (hunk 1 above)
Comment (twokul) [reply lib/broccoli/default-packager.js]: it should have been removed in the first place, and I forgot to do it; unnecessary parameter
